### DISCUSSION TITLE
What is the drawing order for DXF file??

### DISCUSSION THREAD

**Discussion by chibai** [mozman/ezdxf#717] [General]
For example, if there're 2 solid hatches with different colors in the same places~~which one is drawing on the top level??

**Comment by mozman** [comment]
The DXF entities are ordered by ascending HANDLE values - in most case the order of creation.

The drawing order can be changed: https://ezdxf.mozman.at/docs/reorder.html

Example: https://github.com/mozman/ezdxf/blob/master/examples/redraw_order.py

**Reply by chibai** [reply]
Thank you for your answering!  I checked the code to find this~
```
handle_mapping = list(layout.get_redraw_order())
if handle_mapping:
    self.draw_entities(
        reorder.ascending(layout, handle_mapping),
        filter_func=filter_func,
    )
```
I have another extreme case......
For example, there're 2 hatches in 1 Insert, I think their order are the same??
But for the Insert.virtual_entities(), "This entities are not stored in the entity database, have no handle and are not assigned to any layout. "
How do I decide the order of these 2 hatches by EZDXF?

**Reply by mozman** [reply]
In `ezdxf` all entities are handled in the order they are stored in the layout (model-space, block, paper-space). In the sense of handle order, the order of these entities is undefined. Autodesk has for sure some rules for this cases but they don't tell it, you have to find it out by yourself.

**Reply by mozman** [reply]
By logic, the virtual entities of INSERT should be created in the order of the definition in the BLOCK layout. I added some features (v0.18.1) to get the correct redraw order from `ezdxf`, but this is not the default setting, you have to use  `Insert.virtual_entities(redraw_order=True)`.

IMPORTANT: in most cases, the order of creation is the wanted ascending redraw order, and reordering is unnecessary and wastes time

**Reply by chibai** [reply]
Cool
